viskores/cont/testing/UnitTestCubicHermiteSpline.cxx

```diff
@@ -0,0 +1,189 @@
+//============================================================================
+//  The contents of this file are covered by the Viskores license. See
+//  LICENSE.txt for details.
+//
+//  By contributing to this file, all contributors agree to the Developer
+//  Certificate of Origin Version 1.1 (DCO 1.1) as stated in DCO.txt.
+//============================================================================
+
+//============================================================================
+//  Copyright (c) Kitware, Inc.
+//  All rights reserved.
+//  See LICENSE.txt for details.
+//
+//  This software is distributed WITHOUT ANY WARRANTY; without even
+//  the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+//  PURPOSE.  See the above copyright notice for more information.
+//============================================================================
+
+#include <random>
+#include <string>
+
+#include <viskores/cont/CubicHermiteSpline.h>
+#include <viskores/cont/ErrorBadValue.h>
+#include <viskores/cont/Invoker.h>
+#include <viskores/cont/testing/Testing.h>
+#include <viskores/worklet/WorkletMapField.h>
+
+namespace
+{
+
+class SplineEvalWorklet : public viskores::worklet::WorkletMapField
+{
+public:
+  SplineEvalWorklet() {}
+
+  using ControlSignature = void(FieldIn param, ExecObject cubicSpline, FieldOut pos);
+  using ExecutionSignature = void(_1, _2, _3);
+  using InputDomain = _1;
+
+  template <typename ParamType, typename CubicSplineType, typename ResultType>
+  VISKORES_EXEC void operator()(const ParamType& param,
+                                const CubicSplineType& spline,
+                                ResultType& pos) const
+  {
+    auto res = spline.Evaluate(param, pos);
+    if (res != viskores::ErrorCode::Success)
+      this->RaiseError("Spline evaluation failed.");
+  }
+};
+
+void CheckEvaluation(const viskores::cont::CubicHermiteSpline& spline,
+                     const viskores::cont::ArrayHandle<viskores::FloatDefault>& params,
+                     const std::vector<viskores::Vec3f>& posAnswer)
+{
+  viskores::cont::Invoker invoke;
+  viskores::cont::ArrayHandle<viskores::Vec3f> posResult;
+
+  invoke(SplineEvalWorklet{}, params, spline, posResult);
+  VISKORES_TEST_ASSERT(test_equal_ArrayHandles(
+    posResult, viskores::cont::make_ArrayHandle(posAnswer, viskores::CopyFlag::Off)));
+}
+
+void CheckEvaluation(const viskores::cont::CubicHermiteSpline& spline,
+                     const std::vector<viskores::FloatDefault>& params,
+                     const std::vector<viskores::Vec3f>& posAnswer)
+{
+  return CheckEvaluation(
+    spline, viskores::cont::make_ArrayHandle(params, viskores::CopyFlag::Off), posAnswer);
+}
+
+//convenience function to save samples for debugging.
+void SaveSamples(viskores::cont::CubicHermiteSpline& spline)
```

Comment:
The function seems to be declared but never used. This is causing a [compiler warning](https://open.cdash.org/viewBuildError.php?type=1&buildid=10385630).